libscript/sources/Builtins.cxx

```diff
@@ -0,0 +1,326 @@
+        throw e;
+    } catch (std::bad_variant_access e) {
+        throw InvalidException("InvalidOperation", (left.type().toString() + "," + right.type().toString()).c_str(),
+                               "Invalid equal to '==' operator between these types.");
+    }
+}
+data::Value builtin_operatorNEq(const data::Value &left, const data::Value &right)
+{    // not equal to operator '!='
+    try {
+        if (std::get<data::BasicType>(left.type()) == data::BasicType::Number &&
+            std::get<data::BasicType>(right.type()) == data::BasicType::Number)
+            return data::Value(std::get<double>(left) != std::get<double>(right));    // perform arithmetic comparison
+        else if (std::get<data::BasicType>(left.type()) == data::BasicType::String &&
+                 std::get<data::BasicType>(right.type()) == data::BasicType::String)
+            return data::Value(std::get<std::string>(left) !=
+                               std::get<std::string>(right));    // perform alphabetical comparison
+        else if (std::get<data::BasicType>(left.type()) == data::BasicType::Boolean &&
+                 std::get<data::BasicType>(right.type()) == data::BasicType::Boolean)
+            return data::Value(std::get<bool>(left) != std::get<bool>(right));    // perform arithmetic comparison
+        else if (std::get<data::BasicType>(left.type()) == data::BasicType::Vec3 &&
+                 std::get<data::BasicType>(right.type()) == data::BasicType::Vec3)
+            return data::Value(std::get<glm::vec3>(left) != std::get<glm::vec3>(right));    // let glm do the comparison
+        else                                                                                // unsupported
+            throw InvalidException("InvalidOperation", (left.type().toString() + "," + right.type().toString()).c_str(),
+                                   "Invalid not equal to '!=' operator between these types.");
+    } catch (InvalidException e) {
+        throw e;
+    } catch (std::bad_variant_access e) {
+        throw InvalidException("InvalidOperation", (left.type().toString() + "," + right.type().toString()).c_str(),
+                               "Invalid not equal to '!=' operator between these types.");
+    }
+}
+data::Value builtin_operatorGt(const data::Value &left, const data::Value &right)
+{    // greater than operator '>'
+    try {
+        if (std::get<data::BasicType>(left.type()) == data::BasicType::Number &&
+            std::get<data::BasicType>(right.type()) == data::BasicType::Number)
+            return data::Value(std::get<double>(left) > std::get<double>(right));    // perform arithmetic comparison
+        else if (std::get<data::BasicType>(left.type()) == data::BasicType::String &&
+                 std::get<data::BasicType>(right.type()) == data::BasicType::String)
+            return data::Value(std::get<std::string>(left) >
+                               std::get<std::string>(right));    // perform alphabetical comparison
```

Comment:
If you use `std::get_if<std::string>` it would return nullptr if the conversion fail, avoiding the try / catch block just to retrow an exception.

Reply:
I'm using the try/catch because any single fail in std::get (caused by unexpected type) should result in stopping execution and unwinding to the top (by throwing a custom exception describing the error).
I could indeed use ifs to check if std::get would fail, and then throw, but I feel like it bloats the code.
What do you think ?